File: src/dataset/cpp/rna-transcription_exercism-toemm/gemini/GeminiFlash_rna_transcription_v3.cpp
#include "rna_transcription.h"

#include <string>

namespace rna_transcription {

char to_rna(char dna) {
    switch (dna) {
        case 'G': return 'C';
        case 'C': return 'G';
        case 'T': return 'A';
        case 'A': return 'U';
        default: return '\0'; // Or throw an exception for invalid input
    }
}

std::string to_rna(std::string dna) {
    std::string rna;
    rna.reserve(dna.length());

    for (char c : dna) {
        rna += to_rna(c);
    }

    return rna;
}

}  // namespace rna_transcription

File: src/dataset/typescript/rna-transcription_exercism-wackerow/gemini/GeminiFlash_rna-transcription_v3.ts
class Transcriptor {
    private static readonly dnaToRnaMap: { [key: string]: string } = {
        "C": "G",
        "G": "C",
        "A": "U",
        "T": "A"
    };

    toRna(dna: string): string {
        let rna = "";
        for (let i = 0; i < dna.length; i++) {
            const nucleotide = dna[i];
            const rnaNucleotide = Transcriptor.dnaToRnaMap[nucleotide];
            if (rnaNucleotide === undefined) {
                throw "Invalid input DNA.";
            }
            rna += rnaNucleotide;
        }
        return rna;
    }
}

export default Transcriptor